scripts/s3-sanity/s3-sanity-test.sh

```diff
@@ -17,81 +18,301 @@
+  access_key=$(echo -e "$output" | tr ',' '\n' | grep "AccessKeyId" | awk '{print $3}')
+  secret_key=$(echo -e "$output" | tr ',' '\n' | grep "SecretKey" | awk '{print $3}')
+  s3iamcli CreateUser -n SanityUserToDeleteAfterUse --access_key "$access_key" --secret_key "$secret_key" --no-ssl 2> /dev/null
+  if [ $? -ne 0 ];then
+    anys3iamclitestsfailed=true
+    restore_config
+    die_with_error "s3iamcli CreateUser failed"
+  fi
 
-read -p "Enter test bucket name: " bucket_name
-read -p "Enter test object name: " object_name
-read -p "Enter test object size in MB: " object_size
+  if [ "$s3cmdtestson" == true ];then
+    TEST_CMD="s3cmd --access_key=$access_key --secret_key=$secret_key -v -c $s3cmdconfigfile"
+    echo -e "\nCreate bucket - 'sanitybucket': "
+    $TEST_CMD mb "s3://sanitybucket"
+    if [ $? -ne 0 ];then
+      echo "Create bucket - 'sanitybucket':  failed"
+      anys3cmdtestsfailed=true
+    fi
 
-echo -e "\n\n***** S3: SANITY TEST *****"
+    # create a test file
+    dd if=/dev/urandom of=$test_file_input bs=5MB count=1
+    if [ $? -ne 0 ];then
+      echo "dd failed"
+     anys3cmdtestsfailed=true
+    fi
+    content_md5_before=$(md5sum $test_file_input | cut -d ' ' -f 1)
+    if [ $? -ne 0 ];then
+      echo "md5sum failed"
+      anys3cmdtestsfailed=true
+    fi
 
-TEST_CMD="s3cmd --access_key=$access_key --secret_key=$secret_key"
+    echo -e "\nUpload '$test_file_input' to 'sanitybucket': "
+    $TEST_CMD put $test_file_input "s3://sanitybucket/SanityObjectToDeleteAfterUse"
+    if [ $? -ne 0 ];then
+      echo "Upload '$test_file_input' to 'sanitybucket': failed"
+      anys3cmdtestsfailed=true
+    fi
 
-echo -e "\nList Buckets: "
-$TEST_CMD ls || { echo "Failed" && exit 1; }
+    echo -e "\nList uploaded SanityObjectToDeleteAfterUse in 'sanitybucket': "
+    $TEST_CMD ls "s3://sanitybucket/SanityObjectToDeleteAfterUse"
+    if [ $? -ne 0 ];then
+      echo "List uploaded SanityObjectToDeleteAfterUse in 'sanitybucket': failed"
+      anys3cmdtestsfailed=true
+    fi
 
-echo -e "\nCreate bucket - '$bucket_name': "
-$TEST_CMD mb "s3://$bucket_name" || { echo "Failed" && exit 1; }
+    echo -e "\nDownload 'SanityObjectToDeleteAfterUse' from 'sanitybucket': "
+    $TEST_CMD get "s3://sanitybucket/SanityObjectToDeleteAfterUse" $test_output_file
+    if [ $? -ne 0 ];then
+      echo "Download 'SanityObjectToDeleteAfterUse' from 'sanitybucket': failed"
+      anys3cmdtestsfailed=true
+    fi
 
-# create a test file
-test_file_input=/tmp/"$object_name".input
-test_output_file=/tmp/"$object_name".out
+    content_md5_after=$(md5sum $test_output_file | cut -d ' ' -f 1)
+    echo -en "\nData integrity check: "
+    [[ $content_md5_before == $content_md5_after ]] && echo 'Passed.'
+    echo -e "\nDelete 'SanityObjectToDeleteAfterUse' from 'sanitybucket': "
+    $TEST_CMD del "s3://sanitybucket/SanityObjectToDeleteAfterUse"
```

Comment:
Make sure object is completely deleted - either using curl / REST / STs - I leave it to you

Reply:
bgdel verification not done (Write TODO)  - Create a seperate JIRA

Reply:
created https://jts.seagate.com/browse/EOS-18464